operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -1,120 +1,393 @@
+    node images.
+  - **Instructions:** See the "IUF Stage Details for COS" section of the [HPE Cray Operating System Installation Guide CSM on HPE Cray EX Systems (S-8025)](https://www.hpe.com/support/ex-S-8025) for references to the installation
+    procedures.
+- **Content:** Third-party programming environment software
+  - **Description:** CPE provides the `install-3p.sh` script to upload third-party programming environment software to Nexus. The programming environment software is used later in the workflow when creating CPE configurations
+    and building CPE images.
+  - **Instructions:** See the "CPE Install and Upgrade Framework usage" section of the [HPE CPE Installation Guide CSM on HPE Cray EX Systems (S-8003)](https://www.hpe.com/support/ex-S-8003) for references to the installation
+    procedures.
+
+1. Follow instructions to install third-party GPU software.
 
     ```bash
-    SITE_VARS_FILE=/etc/cray/upgrade/csm/iuf/site_vars.yaml
-    vi "${SITE_VARS_FILE}"
+    iuf -a "${ACTIVITY_NAME}" activity --create --comment "installing third-party GPU content" in_progress
+    < perform necessary operations per COS documentation >
+    iuf -a "${ACTIVITY_NAME}" activity --create waiting_admin
     ```
 
-1. Invoke `iuf run` to create an activity named `activity-all-products-upgrade` and use `-e` to execute the [`process-media`](../stages/process_media.md), [`pre-install-check`](../stages/pre_install_check.md),
-   [`deliver-product`](../stages/deliver_product.md), and [`update-vcs-config`](../stages/update_vcs_config.md) stages. Perform the upgrade using product content found in the `${ACTIVITY_DIR}` directory. Use site variables
-   from the `${SITE_VARS_FILE}` file and recipe variables from the `product_vars.yaml` file found in the `${ACTIVITY_DIR}/hpc-csm-software-recipe-23.05.0/vcs` `sat bootprep` configuration directory.
+1. Follow instructions to install third-party programming environment software.
+
+    ```bash
+    iuf -a "${ACTIVITY_NAME}" activity --create --comment "installing third-party programming environment content" in_progress
+    < perform necessary operations per CPE documentation >
+    iuf -a "${ACTIVITY_NAME}" activity --create waiting_admin
+    ```
+
+Once this step has completed:
+
+- Third-party software has been uploaded to Nexus
 
-    (`ncn-m001#`) Execute the stages beginning with `process-media` and ending with `update-vcs-config`.
+## Configuration
+
+This section ensures product configuration have been defined, customized, and is available for later steps in the workflow.
+
+This workflow uses `${ADMIN_DIR}` to retain files that define site preferences for IUF. `${ADMIN_DIR}` is defined separately from `${ACTIVITY_DIR}` and `${MEDIA_DIR}` based on the assumption that the files in `${ADMIN_DIR}` will be
+used when performing future IUF operations unrelated to this workflow.
```

Comment:
It seems like the admin dir we have is a combination of site specific options plus the software recipe.  Do we need an explicit call out to clone to software recipe?